### PR TITLE
bpo-39759: fixed os.getenv docs

### PR DESCRIPTION
**Pull request by dorosch**
https://bugs.python.org/issue39759

### CODE REVIEW

File: Doc/library/os.rst [hunk 1]
```diff
@@ -217,7 +217,8 @@ process and user.
 .. function:: getenv(key, default=None)
 
    Return the value of the environment variable *key* if it exists, or
-   *default* if it doesn't. *key*, *default* and the result are str.
+   *default* if it doesn't. Only *key* needs to be a string as it is used to
```
Comment (remilapeyre) [inline Doc/library/os.rst]: Thanks @dorosch for improving this!

I think we should use `*key* must be a string.`. The "Only" does not make sense if you don't know the previous phrasing and "as it is used to lookup the value in os.environ." is an implementation detail that is not relevant here.

Comment (dorosch) [reply Doc/library/os.rst]: Thanks for review @remilapeyre 
Great point, I'll fix it.

**Review by remilapeyre — APPROVED** [review]
Awesome, thanks!

**Review by serhiy-storchaka — COMMENTED** [review]
Please fix also the documentation for `getenvb`.

**Comment by slateny** [comment]
@dorosch Would you still be interested in updating this PR per Serhiy's comment?

**Comment by serhiy-storchaka** [comment]
The proposed change removes a useful information: that the result is a string if the key is found (and it is a bytes object in `os.getenvv()`). This information should be preserved.

Also, the docstrings say the same as the documentation, so it should be updated correspondingly.

**Comment by slateny** [comment]
Will start on a new PR here then, let me know if you'd still like to work on this one.

**Comment by slateny** [comment]
See #98113

**Pull request closed without merging**